### PR TITLE
Shell: Use shutil.which on Windows

### PR DESCRIPTION
**Pull request by userzimmermann**
@sils @justuswilhelm Continuation of https://github.com/coala/coala/pull/3678

### CODE REVIEW

File: coalib/misc/Shell.py [hunk 1]
```diff
@@ -51,15 +53,18 @@ def run_interactive_shell_command(command, **kwargs):
                     command.
     """
     if not kwargs.get('shell', False) and isinstance(command, str):
-        command = shlex.split(command)
+        command, *cargs = shlex.split(command)
```
Comment (userzimmermann) [inline coalib/misc/Shell.py]: @justuswilhelm Changes type of `command` from sequence to `str`, but only if `shell is True`, so messing up the Windows logic below...

File: coalib/misc/Shell.py [hunk 2]
```diff
@@ -52,6 +54,11 @@ def run_interactive_shell_command(command, **kwargs):
     """
     if not kwargs.get('shell', False) and isinstance(command, str):
         command = shlex.split(command)
+    else:
+        command = list(command)
+
+    if sys.platform.startswith('win'):  # pragma: no cover
```
Comment (Makman2) [inline coalib/misc/Shell.py]: I think you should place a comment above this stmt explaining why we do this
\+
Better use `if platform.system() == 'Windows'` :+1:

Comment (userzimmermann) [inline coalib/misc/Shell.py]: Thanks! Didn't know about `platform.system`. Really the better solution :)

**CC** @justuswilhelm

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/3679/commits/bdde582835a73586589c591d9aab5d21b2b72466

**Comment by userzimmermann** [comment]
unack https://github.com/coala/coala/pull/3679/commits/bdde582835a73586589c591d9aab5d21b2b72466

@Makman2 ack it please :)

**Comment by Makman2** [comment]
ack bdde582

**Comment by userzimmermann** [comment]
@rultor merge

File: coalib/misc/Shell.py [hunk 3]
```diff
@@ -52,6 +54,13 @@ def run_interactive_shell_command(command, **kwargs):
     """
     if not kwargs.get('shell', False) and isinstance(command, str):
         command = shlex.split(command)
+    else:
+        command = list(command)
+
+    if platform.system() == 'Windows':  # pragma: no cover
```
Comment (jayvdb) [inline coalib/misc/Shell.py]: please do not use `# pragma: no cover` for OS-specific code.  At least use `# pragma (posix|nt): no cover` .

**Pull request merged**